delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/StoragesStatistics.scala

```diff
@@ -1,49 +1,92 @@
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.model.StorageStatEntry
 import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.sdk.http.HttpClient.HttpResult
 import ch.epfl.bluebrain.nexus.delta.sdk.model.IdSegment
+import ch.epfl.bluebrain.nexus.delta.sdk.syntax._
 import ch.epfl.bluebrain.nexus.delta.sourcing.model.ProjectRef
+import com.typesafe.scalalogging.Logger
+import io.circe.{Json, JsonObject}
 import monix.bio.{IO, UIO}
 
 trait StoragesStatistics {
 
   /**
-    * Retrieve the current statistics for all storages
-    */
-  def get(): UIO[StorageStatsCollection]
-
-  /**
-    * Retrieve the current statistics for storages in the given project
-    */
-  def get(project: ProjectRef): UIO[Map[Iri, StorageStatEntry]]
-
-  /**
-    * Retrieve the current statistics for the given project
+    * Retrieve the current statistics for a given storage in the given project
     */
   def get(idSegment: IdSegment, project: ProjectRef): IO[StorageFetchRejection, StorageStatEntry]
 
 }
 
-// TODO implements storage statistics
 object StoragesStatistics {
 
-  def apply(): StoragesStatistics = new StoragesStatistics {
+  implicit private val classLoader: ClassLoader = getClass.getClassLoader
+  implicit private val logger: Logger           = Logger[StoragesStatistics]
+
+  private val storagesAggByIdAndProjectQuery =
+    ioJsonObjectContentOf(
+      "query/storages-statistics-aggregations-by-id-proj.json",
+      "project" -> "{{project}}",
+      "storage" -> "{{storage}}"
+    )
+      .logAndDiscardErrors("Storage 'storages-statistics-aggregations-by-id-proj.json' template not found")
+      .memoizeOnSuccess
+
+  /**
+    * @param storageId
+    *   the id of the storage on which to restrict the query
+    * @param projectRef
+    *   the project on which to restrict the query
+    * @return
+    *   an ES query as JsonObject that gets statistics for the given storage in the given project
+    */
+  def statsByIdAndProjectQuery(storageId: Iri, projectRef: ProjectRef): UIO[JsonObject] =
+    storagesAggByIdAndProjectQuery
+      .map(jsonObject =>
+        jsonObject
+          .replace("project" -> "{{project}}", projectRef)
+          .replace("storage" -> "{{storage}}", storageId)
+      )
+
+  def apply(client: ElasticSearchClient, storages: Storages): StoragesStatistics =
+    apply(
+      client.search(_, Set(EventMetricsProjection.eventMetricsIndex.value), Query.Empty)(),
+      storages.fetch(_, _).map(_.id)
+    )
+
+  def apply(
+      client: ElasticSearchClient,
+      fetchStorageId: (IdSegment, ProjectRef) => IO[StorageFetchRejection, Iri]
+  ): StoragesStatistics =
+    apply(
+      client.search(_, Set(EventMetricsProjection.eventMetricsIndex.value), Query.Empty)(),
+      fetchStorageId
+    )
 
-    override def get(): UIO[StorageStatsCollection] = UIO.pure(StorageStatsCollection.empty)
+  def apply(
```

Comment:
We should not need this.
`StorageStatistics` is a SAM (Single Abstract Method) now, we should take profit of it